### PR TITLE
Reformat typography tokens

### PR DESCRIPTION
**Pull request by alexandra-lim**
This MR reformats the typography tokens by creating reusable tokens for all the typography properties such as font family, font size, etc., and expands them in `tokens.json` to allow Style Dictionary to compile them into actual SCSS variables. Cath and I discovered that the [tokens-transformer npm package](https://www.npmjs.com/package/token-transformer) does the same thing, but I'd already reformatted it manually.

This also adds a custom filter to filter for typography types in the tokens as the new format isn't compatible with the base Style Dictionary filter, which uses lodash.

Resources:
- [How to set up custom filters, transforms, etc.](https://github.com/amzn/style-dictionary/issues/94#issuecomment-393018279)
- [registerFilter example](https://github.com/amzn/style-dictionary/issues/303)
- [Stack overflow q re: getting SCSS from design tokens](https://stackoverflow.com/questions/71733046/get-scss-from-design-tokens-json-file-using-style-dictionary)

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const StyleDictionary = require("style-dictionary").extend("config.json");
+
+StyleDictionary.registerFilter({
+  name: "filter-typography",
+  matcher: ({ attributes }) => {
+    return attributes.type === "typography";
+  },
+});
+
+StyleDictionary.extend({
```
Comment (alexandra-lim) [inline build.js:10]: Extending the `config.json` file here trumps what's in `config.json`. I attempted splitting the platforms so the color config was still in `config.json`, but it ended up not transforming those tokens.

File: package.json [hunk 2]
```diff
@@ -3,9 +3,11 @@
   "version": "0.0.1",
   "description": "Testing 123",
   "main": "index.js",
-  "files": ["index.js"],
+  "files": [
+    "index.js"
+  ],
   "scripts": {
-    "build": "style-dictionary build"
+    "build": "node build.js"
```
Comment (alexandra-lim) [inline package.json:10]: Points to the new build.js to run the custom configs

File: build/ts/typography.ts [hunk 3]
```diff
@@ -1,13 +1,69 @@
...
-export const GlobalTypographyH4 = {"fontFamily":"Mulish","fontWeight":"700","lineHeight":"24","fontSize":"20","letterSpacing":"0%","paragraphSpacing":"0","textDecoration":"none","textCase":"none"};
-export const GlobalTypographyH5 = {"fontFamily":"Mulish","fontWeight":"700","lineHeight":"24","fontSize":"16","letterSpacing":"0%","paragraphSpacing":"0","textDecoration":"none","textCase":"none"};
-export const GlobalTypographyBody = {"fontFamily":"Mulish","fontWeight":"500","lineHeight":"24","fontSize":"16","letterSpacing":"0%","paragraphSpacing":"0","textDecoration":"none","textCase":"none"};
-export const GlobalTypographyFootnote = {"fontFamily":"Mulish","fontWeight":"500","lineHeight":"16","fontSize":"14","letterSpacing":"0%","paragraphSpacing":"0","textDecoration":"none","textCase":"none"};
-export const GlobalTypographyCaption = {"fontFamily":"Mulish","fontWeight":"500","lineHeight":"16","fontSize":"12","letterSpacing":"0%","paragraphSpacing":"0","textDecoration":"none","textCase":"none"};
\ No newline at end of file
+export const GlobalTypographyH1FontFamily = "Mulish";
```
Comment (alexandra-lim) [inline build/ts/typography.ts:6]: This new format is not an object like previously generated, but matches the `typography.ts` file we have in `hw-admin`. @iryanclarke, is this the format you had in mind?

**Comment by CCAyl** [comment]
I'm going to close my open PR, it will just conflict with this guy and this has much more valuable stuff in it. the folder structure is easy to change

**Comment by CCAyl** [comment]
this LGTM, we can still play around with transformers etc, but having this new build.js is key

**Review by iryanclarke — COMMENTED** [review]
Just the one main comment :)

File: build/scss/typography.scss [hunk 4]
```diff
@@ -1,12 +1,68 @@
...
+$global-typography-h5-line-height: 24;
+$global-typography-h5-font-size: 16;
+$global-typography-h5-letter-spacing: 0;
+$global-typography-h5-paragraph-spacing: 0;
+$global-typography-h5-text-decoration: none;
+$global-typography-h5-text-case: none;
+$global-typography-body-font-family: Mulish;
+$global-typography-body-font-weight: 500;
+$global-typography-body-line-height: 24;
+$global-typography-body-font-size: 16;
```
[the remark this answers is not in the document]
Comment (alexandra-lim) [reply build/scss/typography.scss:48]: Yeah! We haven't included it as part of the values from the get-go. We can add it to the values directly, or we can leverage this[ custom transformer ](https://github.com/six7/figma-tokens/issues/703#issuecomment-1130246775) to add the units as a suffix. My plan was to add the custom transformer to keep the values as a numerical value only

[the remark this answers is not in the document]
Comment (iryanclarke) [reply build/scss/typography.scss:48]: Ok sounds good then!

**Review by iryanclarke — APPROVED** [review]

**Pull request merged**